includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1270,6 +1270,8 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 			$payment_method_details = $charge ? $charge->get_payment_method_details() : [];
 			$payment_method_type    = $payment_method_details ? $payment_method_details['type'] : null;
 
+			$this->order_service->attach_transaction_fee_to_order( $order, $charge );
```

Comment:
I guess it was placed there  because existing `$charge` object was being retrieved there. I moved it now, and all scenarios seem to work.
The new position does not help with 3DS though. 3DS with UPE works, possibly because of the code in the redirect methods. Non-UPE checkout + 3DS does not work either ways. Will look for another solution for that.

Reply:
The 3DS + non-UPE orders gets updated with `_charge_id` only during webhook processing, so I think we should be OK with the similar solution for `_wcpay_transaction_fee` as well.

Reply:
Ah, I see now. For 3DS + non-UPE flow, before we re-direct to the order-received page, we send a AJAX request and it's handled in `WC_Payment_Gateway_WCPay::update_order_status`, so I think putting it inside this handling would be enough: 

https://github.com/Automattic/woocommerce-payments/blob/50ab8e33720c764c2e9a75b094083ae581b1770a/includes/class-wc-payment-gateway-wcpay.php#L2745-L2746